druntime/src/core/internal/gc/impl/conservative/gc.d

```diff
@@ -3662,7 +3682,7 @@ Lmark:
         if (!scanThreadData)
             onOutOfMemoryError();
 
-        evStart.initialize(false, false);
+        evStackFilled.initialize(false, false);
         evDone.initialize(false, false);
```

Comment:
I think we can completely remove evDone. It's only waited on by the main thread with a 1ms timeout when the GC is shutting down. Could either replace with a sleep, or just remove this whole "exit loop" altogether -- if you set the stopGC flag, have the background threads set the evStackFilled event on exit, to wake up the next one.

Then the main thread will sleep on joining the background threads, and that is the true end of the loop.

Reply:
I thought so too, but wasn't sure whether that can fail and never terminate. IIRC there can be spurious wake-ups when waiting for events, maybe a wake-up might also be missed sometimes? Maybe an unconditional sleep is a simple and safe version.